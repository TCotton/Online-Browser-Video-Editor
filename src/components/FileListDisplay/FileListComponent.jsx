import React, { useState, useEffect } from 'react'
import { useLiveQuery } from 'dexie-react-hooks'
import dbVF from '../indexDB/indexDBVideo'
import { FileListComponents } from './FileListComponents'

export const FileListComponent = () => {
  const [videoFile, setVideoFile] = useState([])
  const allItems = useLiveQuery(() => dbVF.videofile.toArray(), [])

  useEffect(() => {
    if (allItems && Array.isArray(allItems)) {
      setVideoFile(allItems)
    }
  }, [allItems])

  return (
    <div className='filelists' data-testid='FileListComponent'>
      <FileListComponents data={videoFile} />
    </div>
  )
}
