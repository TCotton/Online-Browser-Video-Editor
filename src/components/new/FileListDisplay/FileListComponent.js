import React from 'react';

export const FileListComponent = () => {
    return (
        <div data-testid='FileListComponent'/>
    )
}