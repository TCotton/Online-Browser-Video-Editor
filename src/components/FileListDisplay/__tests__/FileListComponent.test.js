import React from 'react'
import { render } from '@testing-library/react'
import { FileListComponent } from '../FileListComponent'

jest.mock('dexie-react-hooks', () => ({
  ...jest.requireActual('dexie-react-hooks'),
  useLiveQuery: jest.fn()
}))

describe('FileListComponent', () => {
  it('should render successfully', () => {
    const { container } = render(<FileListComponent />)
    expect(
      container.querySelector('[data-testid="FileListComponent"]')
    ).not.toBeNull()
  })

  it('should match snapshot', () => {
    const { baseElement } = render(<FileListComponent />)
    expect(baseElement).toMatchSnapshot()
  })
})
