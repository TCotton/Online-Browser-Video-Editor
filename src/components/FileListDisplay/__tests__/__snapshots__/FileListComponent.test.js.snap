// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileListComponent should match snapshot 1`] = `
<body>
  <div>
    <div
      class="filelists"
      data-testid="FileListComponent"
    >
      <section
        class="file-row"
        data-testid="FileListComponents"
      />
    </div>
  </div>
</body>
`;
